src/main/java/com/google/api/codegen/util/CommonAcronyms.java

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.api.codegen.util;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Utility class to replace fully capitalized common acronyms with an upper camel interpretation.
+ */
+public class CommonAcronyms {
+  private static final ImmutableMap<String, String> ACRONYMS =
+      ImmutableMap.<String, String>builder()
+          .put("IAM", "Iam")
+          .put("HTTP", "Http")
+          .put("XML", "Xml")
+          .put("API", "Api")
+          .build();
+
+  public static String replaceAcronyms(String str) {
+    for (String acronym : ACRONYMS.keySet()) {
+      str = str.replace(acronym, ACRONYMS.get(acronym));
+    }
```

Comment:
Agreed. I may want to add some documentation clearly stating that this case is not covered and will probably cause an error.